substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/config/SerializationConfigurationLambdaCapturingType.java

```diff
@@ -0,0 +1,83 @@
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.configure.config;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.graalvm.nativeimage.impl.ConfigurationCondition;
+
+import com.oracle.svm.configure.json.JsonWriter;
+import com.oracle.svm.core.configure.SerializationConfigurationParser;
+
+public class SerializationConfigurationLambdaCapturingType extends SerializationConfigurationAbstractType implements Comparable<SerializationConfigurationLambdaCapturingType> {
+    private final ConfigurationCondition condition;
+    private final String qualifiedJavaName;
+
+    public SerializationConfigurationLambdaCapturingType(ConfigurationCondition condition, String qualifiedJavaName) {
+        assert qualifiedJavaName.indexOf('/') == -1 : "Requires qualified Java name, not internal representation";
+        assert !qualifiedJavaName.startsWith("[") : "Requires Java source array syntax, for example java.lang.String[]";
+        Objects.requireNonNull(condition);
+        this.condition = condition;
+        Objects.requireNonNull(qualifiedJavaName);
+        this.qualifiedJavaName = qualifiedJavaName;
+    }
+
+    @Override
+    public void printJson(JsonWriter writer) throws IOException {
+        writer.append('{').indent().newline();
+        ConfigurationConditionPrintable.printConditionAttribute(condition, writer);
+
+        writer.quote(SerializationConfigurationParser.LAMBDA_CAPTURING_CLASS_KEY).append(":").quote(qualifiedJavaName);
+        writer.unindent().newline().append('}');
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        SerializationConfigurationLambdaCapturingType that = (SerializationConfigurationLambdaCapturingType) o;
+        return condition.equals(that.condition) &&
+                        qualifiedJavaName.equals(that.qualifiedJavaName);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(condition, qualifiedJavaName);
+    }
+
+    @Override
+    public int compareTo(SerializationConfigurationLambdaCapturingType other) {
```

Comment:
How is sorting done between the two types?

Reply:
New comparator in super class has been intorduced. It only compares by subtype, and if subtypes of the two objects are the same, then we compare order them by the values of their fields.